cmd/operator-sdk/bundle/validate.go

```diff
@@ -102,63 +129,80 @@ func (c bundleValidateCmd) validate(args []string) error {
 	return nil
 }
 
+// TODO: add a "permissive" flag to toggle whether warnings also cause a non-zero
+// exit code to be returned (true by default).
 func (c *bundleValidateCmd) addToFlagSet(fs *pflag.FlagSet) {
 	fs.StringVarP(&c.imageBuilder, "image-builder", "b", "docker",
-		"Tool to extract container images. One of: [docker, podman]")
+		"Tool to extract bundle image data. Only used when validating a bundle image. "+
+			"One of: [docker, podman]")
 }
 
-func (c bundleValidateCmd) run() (err error) {
-	// Set directory, either supplied directly or a temp dir used to unpack
-	// the image.
-	dir := c.directory
-	if c.imageTag != "" {
-		dir, err = ioutil.TempDir("", "bundle-")
-		if err != nil {
-			return err
-		}
-		defer func() {
-			if err = os.RemoveAll(dir); err != nil {
-				log.Errorf("Error removing temp bundle dir: %v", err)
-			}
-		}()
-	}
-	if dir, err = filepath.Abs(dir); err != nil {
-		return err
-	}
+func (c bundleValidateCmd) run() error {
+
+	logger := log.WithFields(log.Fields{
+		"bundle-dir":     c.directory,
+		"container-tool": c.imageBuilder,
+	})
+	val := registrybundle.NewImageValidator(c.imageBuilder, logger)
 
-	// Set up logger.
-	fields := log.Fields{"bundle-dir": dir}
-	if c.imageTag != "" {
-		fields["container-tool"] = c.imageBuilder
+	// Validate bundle format.
+	if err := val.ValidateBundleFormat(c.directory); err != nil {
+		return fmt.Errorf("invalid bundle format: %v", err)
 	}
-	logger := log.WithFields(fields)
-	if viper.GetBool(flags.VerboseOpt) {
-		log.SetLevel(log.DebugLevel)
+
+	// Validate bundle content.
+	// TODO(estroz): instead of using hard-coded 'manifests', look up bundle
+	// dir name in metadata labels.
+	manifestsDir := filepath.Join(c.directory, registrybundle.ManifestsDir)
+	if results, err := validateBundleContent(logger, manifestsDir); err != nil {
+		return fmt.Errorf("error validating %s: %v", manifestsDir, err)
+	} else if checkResults(results) {
+		return errors.New("invalid bundle content")
```

Comment:
Nit: Can we not use `else if`. 
I find that falling through to the second condition makes the code more readable.
```suggestion
	results, err := validateBundleContent(logger, manifestsDir)
	if err != nil {
		return fmt.Errorf("error validating %s: %v", manifestsDir, err)
	}
	if checkResults(results) {
		return errors.New("invalid bundle content")
```